src/components/Feed/Feed.test.tsx

```diff
@@ -0,0 +1,196 @@
+jest.mock("../../hooks/useStories");
+
+const mockedUseStories = useStories as jest.Mock;
+
+describe("add list button component", () => {
+  beforeEach(() => {
+    mockedUseStories.mockClear();
+    mockedUseStories.mockReturnValue({
+      stories: stories,
+      loading: false,
+      loadMore: jest.fn(),
+      hasMore: false,
+      errors: [],
+    });
+  });
+
+  it("renders chips and selects the first by default", () => {
+    render(
+      <MemoryRouter>
+        <Feed />
+      </MemoryRouter>
+    );
+    storyTypeChips.forEach((chip) => {
+      expect(screen.getByTestId(`chip-${chip.name}`)).toBeInTheDocument();
+    });
+  });
+
+  it("renders NewsCard components for each story with correct test IDs", async () => {
+    mockedUseStories.mockReturnValue({
+      stories: stories,
+      loading: false,
+      loadMore: jest.fn(),
+      hasMore: false,
+      errors: [],
+    });
+
+    render(
+      <MemoryRouter>
+        <Feed />
+      </MemoryRouter>
+    );
+
+    for (const story of stories) {
+      const card = await screen.findByTestId(`news-card-${story.id}`);
+      expect(card).toBeInTheDocument();
+      expect(card).toHaveTextContent(story.title);
+    }
+  });
+
+  it("shows loading indicator when loading is true", () => {
+    mockedUseStories.mockReturnValue({
+      stories: [],
+      loading: true,
+      loadMore: jest.fn(),
+      hasMore: false,
+      errors: [],
+    });
+
+    render(
+      <MemoryRouter>
+        <Feed />
+      </MemoryRouter>
+    );
+
+    const loader = screen.getByTestId("loading-indicator");
+    expect(loader).toBeInTheDocument();
+  });
+
+  it("renders Load More button when there are more stories and not loading", async () => {
+    const loadMore = jest.fn();
+
+    mockedUseStories.mockReturnValue({
+      stories: stories,
+      loading: false,
+      loadMore,
+      hasMore: true,
+      errors: [],
+    });
+
+    render(
+      <MemoryRouter>
+        <Feed />
+      </MemoryRouter>
+    );
+
+    const loadMoreButton = screen.getByText("Load More");
+    expect(loadMoreButton).toBeInTheDocument();
+
+    await userEvent.click(loadMoreButton);
+    expect(loadMore).toHaveBeenCalled();
+  });
+
+  it("shows Load More button when hasMore is true and not loading", async () => {
+    const loadMore = jest.fn();
+    mockedUseStories.mockReturnValue({
+      stories: [{ id: 1, title: "More News" }],
+      loading: false,
+      loadMore,
+      hasMore: true,
+      errors: [],
+    });
+
+    render(
+      <MemoryRouter>
+        <Feed />
+      </MemoryRouter>
+    );
+    const button = screen.getByText("Load More");
+    expect(button).toBeInTheDocument();
+
+    await userEvent.click(button);
+    expect(loadMore).toHaveBeenCalled();
+  });
+
+  it("changes chip selection and triggers hook again", async () => {
+    mockedUseStories.mockReturnValue({
+      stories: [],
+      loading: false,
+      loadMore: jest.fn(),
+      hasMore: false,
+      errors: [],
```

Comment:
Create a common constant for all these values and update the changed values only.